lib/http.js

```diff
@@ -0,0 +1,45 @@
+import { getLatestMetrics, followMetrics } from '../lib/metrics.js'
+import http from 'node:http'
+import { once } from 'node:events'
+
+const asyncHandler = async (req, res) => {
+  const url = new URL(req.url, 'http://localhost/')
+
+  if (url.pathname === '/metrics') {
+    res.setHeader('Content-Type', 'application/json')
+    if (url.searchParams.has('follow')) {
+      const controller = new AbortController()
+      const { signal } = controller
+      await Promise.all([
+        (async () => {
+          try {
+            await once(req, 'close')
+          } catch {
+          } finally {
+            controller.abort()
+          }
```

Comment:
This is a lot of code, mostly keywords and syntax tokens. Have you considered rewriting this using Promise API?

```js
once(req, 'close').then(
  _ok => controller.abort(),
  _err => controller.abort(),
);
````

Reply:
I find this cleaner, as it's using the core language constructs `try` / `catch` and `finally` (proven from most languages), instead of the much debated Promise api. I understand it's more verbose